File: contactlist/src/main/java/com/djamil/contactlist/ContactList.java
package com.djamil.contactlist;

import android.app.Activity;
import android.content.Intent;

import com.djamil.contactlist.interfaces.OnClickContactListener;

/**
 * @Author Moustapha S. Dieme ( Djvmil_ ) on 10/12/19.
 */

public class ContactList {
    private static ContactList instance;
    private static Activity activity;
    private static ContactListActivity contactListActivity;
    static Boolean multiContact = true;
    static int limit;
    static OnClickContactListener onClickContactListener;

    private ContactList() {

    }

    public static ContactList getInstance(Activity act) {
        activity = act;
        contactListActivity = new ContactListActivity();
        if (instance == null)
            instance = new ContactList();

        return instance;
    }

    public void showContactList(){
        activity.startActivity(new Intent(activity, contactListActivity.getClass()));
    }

    public void showContactList(Boolean multipleContact, int limitContact){
        multiContact = multipleContact;
        limit = limitContact;
        activity.startActivity(new Intent(activity, contactListActivity.getClass()));
    }

    public void setOnClickContactListener(OnClickContactListener onClickListener) {
        onClickContactListener = onClickListener;
    }

    public void removeInstance() {
        activity = null;
        instance = null;
    }

}

File: contactlist/src/main/java/com/djamil/contactlist/ContactList.kt
package com.djamil.contactlist

import android.app.Activity
import android.content.Intent
import com.djamil.contactlist.ContactList
import com.djamil.contactlist.interfaces.OnClickContactListener
import com.djamil.contactlist.ContactListActivity

/**
 * @Author Moustapha S. Dieme ( Djvmil_ ) on 10/12/19.
 */
class ContactList private constructor() {
    fun showContactList() {
        activity!!.startActivity(Intent(activity, contactListActivity!!.javaClass))
    }

    fun showContactList(multipleContact: Boolean = false, limitContact: Int = -1, msgLimitContact: String = "") {
        multiContact = multipleContact
        limit = limitContact
        msgLimit = msgLimitContact
        activity!!.startActivity(Intent(activity, contactListActivity!!.javaClass))
    }

    fun setOnClickContactListener(onClickListener: OnClickContactListener?) {
        onClickContactListener = onClickListener
    }

    fun removeInstance() {
        activity = null
        instance = null
        contactListActivity?.finish()
        contactListActivity = null
    }

    companion object {
        private var activity: Activity? = null
        private var contactListActivity: ContactListActivity? = null
        private var instance: ContactList? = null
        var multiContact = false
        var limit = -1
        var msgLimit = ""
        var onClickContactListener: OnClickContactListener? = null

        @JvmStatic
        fun getInstance(act: Activity?): ContactList {
            activity = act
            contactListActivity = ContactListActivity()
            if (instance == null) instance = ContactList()
            return instance!!
        }
    }
}